kubetest2-gke/deployer/firewall.go

```diff
@@ -91,17 +93,20 @@ func (d *deployer) cleanupNetworkFirewalls(project, network string) (int, error)
 		klog.V(1).Infof("Network %s has %v undeleted firewall rules %v", network, len(fwList), fwList)
 		commandArgs := []string{"compute", "firewall-rules", "delete", "-q"}
 		commandArgs = append(commandArgs, fwList...)
-		commandArgs = append(commandArgs, "--project="+project)
+		commandArgs = append(commandArgs, "--project="+hostProject)
 		errFirewall := runWithOutput(exec.Command("gcloud", commandArgs...))
 		if errFirewall != nil {
 			return 0, fmt.Errorf("error deleting firewall: %v", errFirewall)
 		}
-		return len(fwList), nil
+		// Sleep 10 seconds to wait for the firewall rules being deleted completely.
+		// TODO(chizhg): change to a more reliable way to check if they are deleted or not.
+		time.Sleep(10 * time.Second)
```

Comment:
does this mean the `gcloud firewall-rules delete` command exits before the firewalls are actually deleted?

Reply:
Based on my experiment, yes, sometimes I got "network is still used by firewall rule xxx", even if the log shows the firewall rule has been deleted.
I have changed the comment to include this information.